pycheribuild/qemu_utils.py

```diff
@@ -88,14 +88,29 @@ def __init__(self, xtarget: CrossCompileTarget, want_debugger=False):
         else:
             raise ValueError("Unknown target " + str(xtarget))
 
-    def disk_image_args(self, image) -> list:
+    def disk_image_args(self, image: Path, image_format: str) -> list:
+        # Probe the disk image format in case someone has overridden the default image path or format is unspecified
+        if not image.exists():
+            # Either we're pretending or we'll complain elsewhere.
+            if image_format is None:
+                image_format = "qcow2" if image.name.endswith(".qcow2") else "raw"
+        else:
+            with image.open('rb') as imgf:
+                magic = imgf.read(4)
+                is_qcow2 = magic == b'QFI\xfb'
+
+                if image_format is None:
+                    image_format = "qcow2" if is_qcow2 else "raw"
+                elif is_qcow2 and image_format != "qcow2":
+                    warning_message("Disk image looks like qcow2 but image_format is", image_format)
```

Comment:
Should also warn for the `not is_qcow2 and image_format == "qcow2"` case, i.e. emit a warning whenever `is_qcow2 != (image_format == "qcow2")`